Type-clean EventQueue successor: abstract base makes EventQueue() unusable for mypy users

## Problem

PR #944 split `EventQueue` into an abstract base + concrete `EventQueueLegacy`, with `EventQueue.__new__` redirecting `EventQueue(...)` to `EventQueueLegacy(...)` at runtime for backwards compatibility. The docstring of `EventQueue` shows `queue = EventQueue()` as typical usage.

However, mypy correctly flags this pattern as an error because `EventQueue` is declared abstract via `ABC` + `@abstractmethod` on `enqueue_event`:

```
error: Cannot instantiate abstract class "EventQueue" with abstract attribute "enqueue_event"  [abstract]
error: "EventQueue" has no attribute "dequeue_event"; maybe "enqueue_event"?  [attr-defined]
```

The methods `dequeue_event`, `tap`, `close`, `task_done`, `is_closed` only exist on `EventQueueLegacy`, not on the abstract `EventQueue` interface.

Result: downstream type-checked codebases (we ship a Python SDK that strict-mode-`mypy`s every commit across 3.10–3.13) must keep importing `EventQueueLegacy` explicitly — the class name advertises its own deprecation, but it remains the only typeable option in a2a-sdk 1.0.1.

## a2a-sdk 1.0.1 internal usage

Even a2a-sdk's own modules never instantiate `EventQueue()` — they all use `EventQueueLegacy()`:

- `a2a/server/events/in_memory_queue_manager.py`: `queue = EventQueueLegacy()`
- `a2a/server/events/queue_manager.py`: typed as `EventQueueLegacy`
- `a2a/server/events/event_consumer.py`: typed as `EventQueueLegacy`
- `a2a/server/request_handlers/default_request_handler.py`: `queue = EventQueueLegacy()`

Only the docstring of `EventQueue` itself shows `queue = EventQueue()`.

## Requested

One of:

1. Promote `EventQueue` to a concrete (non-abstract) class that owns the full method surface (`dequeue_event`, `tap`, `close`, `task_done`, `is_closed`), making `EventQueueLegacy` a thin alias for the deprecation window.
2. Add a `DefaultEventQueue` (or similar) concrete class that downstream users can import and instantiate without the `*Legacy` name, with the same method surface as `EventQueueLegacy`.
3. Document that `EventQueueLegacy` is the supported instantiation surface for the foreseeable future and update the `EventQueue` docstring's `queue = EventQueue()` example accordingly so users (and type-checkers) aren't misled.

We don't have a preference on which path — any of the three unblocks the rename downstream.

## Downstream tracking

adcontextprotocol/adcp-client-python#699 (left a TODO in our harness pointing at this issue).

--- a/src/a2a/server/events/event_queue.py
+++ b/src/a2a/server/events/event_queue.py
@@ -1,6 +1,7 @@
 import asyncio
 import logging
 import sys
+import warnings
 
 from abc import ABC, abstractmethod
 from types import TracebackType
@@ -48,38 +49,28 @@ DEFAULT_MAX_QUEUE_SIZE = 1024
 
 
 class EventQueue(ABC):
-    """Base class and factory for EventQueueSource.
+    """Producer-side interface passed to `AgentExecutor.execute`/`cancel`.
 
-    EventQueue provides an abstraction for a queue of events that can be tapped
-    by multiple consumers.
-    EventQueue maintain main queue and source and maintain child queues in sync.
-    GUARANTEE: All sinks (including the default one) will receive events in the exact same order.
+    Exposes only `enqueue_event`. The consumer is framework-managed
+    and not part of the public surface.
 
-    WARNING (Concurrency): All events from all sinks (both the default queue and any
-    tapped child queues) must be regularly consumed and marked as done. If any single
-    consumer stops processing and its queue reaches capacity, it can block the event
-    dispatcher and stall the entire system, causing a widespread deadlock.
-
-    WARNING (Memory Leak): Event queues spawn background tasks. To prevent memory
-    and task leaks, all queue objects (both source and sinks) MUST be explicitly
-    closed via `await queue.close()` or by using the async context manager (`async with queue:`).
-    Child queues are automatically closed when parent queue is closed, but you
-    should still close them explicitly to prevent queues from reaching capacity by
-    unconsumed events.
-
-    Typical usage:
-    queue = EventQueue()
-    child_queue1 = await queue.tap()
-    child_queue2 = await queue.tap()
-
-    async for event in child_queue1:
-        do_some_work(event)
-        child_queue1.task_done()
+    Default request handlers construct the queue and pass it in; executors
+    should accept it and not construct one. To run an executor outside the
+    framework, write a custom subclass or use `EventQueueLegacy` (deprecated,
+    will be removed in a future release).
     """
 
     def __new__(cls, *args: Any, **kwargs: Any) -> Self:
-        """Redirects instantiation to EventQueueLegacy for backwards compatibility."""
+        """Deprecated: redirects bare `EventQueue(...)` to `EventQueueLegacy(...)`."""
         if cls is EventQueue:
+            warnings.warn(
+                'EventQueue is an abstract interface; instantiating it '
+                'directly is deprecated. The redirect to EventQueueLegacy '
+                '(and EventQueueLegacy itself) will be removed in a future '
+                'release.',
+                DeprecationWarning,
+                stacklevel=2,
+            )
             instance = EventQueueLegacy.__new__(EventQueueLegacy)
             EventQueueLegacy.__init__(instance, *args, **kwargs)
             return cast('Self', instance)

